Application.cpp

```diff
@@ -120,6 +120,7 @@ bool Application::onInit() {
 	SupportedLimits supportedLimits;
 	adapter.getLimits(&supportedLimits);
 	RequiredLimits requiredLimits = Default;
+	requiredLimits.limits = supportedLimits.limits;
```

Comment:
Don't blindly forward the adapter's limits, do it only for the fields that really need to be forwarded but most of the limits should be explicit. Only `minStorageBufferOffsetAlignment` here may be forwarded.

Note that the limit issue is not specific to macOS, it is due to a change in the `wgpu-native` backend, that is now more strict (as it should be) about these limits. I've started updating all chapters but it takes time; as of now I did it up to [`step044`](https://github.com/eliemichel/LearnWebGPU-Code/blob/step044/main.cpp#L90).

Reply:
I needed it for `minUniformBufferOffsetAlignmenrequired` and 
`minStorageBufferOffsetAlignment`, I guess my idea of just starting out with the defaults wasn't great (the intention was to avoid other similar issues).
I'll update the PR to explicitly set only those two.

Reply:
Yes only the 'min' ones should be copied this way. See page 66-68 of [this](https://eliemichel.github.io/LearnWebGPU/_downloads/e3781dfa813253f88fdd839549e8339c/2023-03_WebGPU%20is%20not%20just%20about%20the%20Web.pdf) for detailed explainations ;)